backend/src/main/java/com/google/rolecall/ApplicationLoader.java

```diff
@@ -61,6 +67,86 @@ private void createAdmin() {
     userRepo.save(admin);
     logger.log(Level.WARNING, String.format("Admin User Created: %s %s %s", 
         adminFirstName, adminLastName, adminEmail));
+    createTestData();
+  }
+
+  // Create sample data
+  
+  private enum DataCreateError {
+    OK,
+    OtherError
+  }
+
+  private DataCreateError createOneUser(String fName, String lName, String email, String dateJoined) {
+    Calendar cal = Calendar.getInstance();
+    SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/yyyy", Locale.US);
+    User user;
+    try {
+      cal.setTime(sdf.parse(dateJoined));
+
+      user = User.newBuilder()
+          .setFirstName(fName)
+          .setLastName(lName)
+          .setEmail(email)
+          .setDateJoined(cal)
+          .setIsActive(true)
+          .setCanLogin(true)
+          .setIsDancer(true)
+          .build();
+      userRepo.save(user);
+    } catch(InvalidParameterException e) {
+      return DataCreateError.OtherError;
+    } catch(ParseException e) {
+      return DataCreateError.OtherError;
+    }
+    return DataCreateError.OK;
+  }
+
+  private DataCreateError createTestData() {
+    logger.log(Level.WARNING, "Creating test data.");    
+    Boolean finished = false;
+    while (!finished) {
```

Comment:
You are right. I shouldn't have captured the error in the 'lower' function and instead handled it in the 'higher' function with the added benefit that the higher level function would be cleaner.

On the other hand, the only way for me to test any code changes here is to wipe the database (to your point below) repeatedly, and right now I have created good demo data.  And, I will try to explain below, this code will have a limited life span (I plan to remove it as soon as Alvin Ailey is up and running).

Rather than fixing the code, I vote that we leave it in for now (it works after all) and remove it in a week or two.

Reply:
I think the fix should be relatively simple.

In createOneUser, let's throw exception if the creation failed.

In createTestData, let use

```
try {
  createOneUser("Robert", "Battle", "rb@gmail.com","1/1/2020"));
  createOneUser("Jeroboam", "Bozeman", "bb@gmail.com","1/1/2020"))
  ...
} catch (SomeException e) {
  // handle the exception such as logging the error
}
```

I am just not in favor of the while loop. Seems it is not the best fit here and we can make it right at the first time.

Reply:
Done.

Reply:
Thanks.